inbm/dispatcher-agent/dispatcher/sota/snapshot.py

```diff
@@ -223,10 +227,13 @@ def recover(
         dispatcher_state.clear_dispatcher_state()
         if self.snap_num:
             self._rollback_and_delete_snap()
+            logger.debug("Rebooting to recover from failed SOTA...")
+            rebooter.reboot()
```

Comment:
**Suggestion:** Remove redundant logging statement before the reboot, as it is already logged within the _rollback_and_delete_snap method. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            if self.snap_num:
                self._rollback_and_delete_snap()
                rebooter.reboot()
```